pandas/tests/io/parser/common/test_chunksize.py

```diff
@@ -162,14 +163,18 @@ def test_chunk_begins_with_newline_whitespace(all_parsers):
 
 
 @pytest.mark.slow
-def test_chunks_have_consistent_numerical_type(all_parsers):
+def test_chunks_have_consistent_numerical_type(all_parsers, monkeypatch):
+    # mainly an issue with the C parser
+    heuristic = 2**3
     parser = all_parsers
-    integers = [str(i) for i in range(499999)]
+    integers = [str(i) for i in range(heuristic - 1)]
```

Comment:
Does this need to expand all the way to 2**20?

Reply:
I think it needs to hit `DEFAULT_BUFFER_HEURISTIC - 1` to make read_csv with the c parser create a buffer with int and float strings

Reply:
My mistake - I missed the monkeypatch here so thought this was doing the whole range in 2**20. This looks good